mm2src/mm2_main/src/rpc/lp_commands/lp_commands_legacy.rs

```diff
@@ -30,52 +30,89 @@ use mm2_metrics::MetricsOps;
 use mm2_number::{construct_detailed, BigDecimal};
 use serde_json::{self as json, Value as Json};
 use std::borrow::Cow;
+use std::iter::Extend;
+use uuid::Uuid;
 
 use crate::mm2::lp_dispatcher::{dispatch_lp_event, StopCtxEvent};
 use crate::mm2::lp_network::subscribe_to_topic;
-use crate::mm2::lp_ordermatch::{cancel_orders_by, CancelBy};
-use crate::mm2::lp_swap::{active_swaps_using_coin, tx_helper_topic, watcher_topic};
+use crate::mm2::lp_ordermatch::{cancel_orders_by, get_matching_orders, CancelBy};
+use crate::mm2::lp_swap::{active_swaps_using_coins, tx_helper_topic, watcher_topic};
 use crate::mm2::MmVersionResult;
 
+const INTERNAL_SERVER_ERROR_CODE: u16 = 500;
+const RESPONSE_OK_STATUS_CODE: u16 = 200;
+
+pub fn disable_coin_err(
+    error: String,
+    matching: &[Uuid],
+    cancelled: &[Uuid],
+    active_swaps: &[Uuid],
+) -> Result<Response<Vec<u8>>, String> {
+    let err = json!({
+        "error": error,
+        "orders": {
+            "matching": matching,
+            "cancelled": cancelled
+        },
+        "active_swaps": active_swaps
+    });
+    Response::builder()
+        .status(INTERNAL_SERVER_ERROR_CODE)
+        .body(json::to_vec(&err).unwrap())
+        .map_err(|e| ERRL!("{}", e))
+}
+
 /// Attempts to disable the coin
 pub async fn disable_coin(ctx: MmArc, req: Json) -> Result<Response<Vec<u8>>, String> {
     let ticker = try_s!(req["coin"].as_str().ok_or("No 'coin' field")).to_owned();
-    let _coin = match lp_coinfind(&ctx, &ticker).await {
+    let coin = match lp_coinfind(&ctx, &ticker).await {
         Ok(Some(t)) => t,
         Ok(None) => return ERR!("No such coin: {}", ticker),
         Err(err) => return ERR!("!lp_coinfind({}): ", err),
     };
-    let swaps = try_s!(active_swaps_using_coin(&ctx, &ticker));
-    if !swaps.is_empty() {
-        let err = json!({
-            "error": format!("There're active swaps using {}", ticker),
-            "swaps": swaps,
-        });
-        return Response::builder()
-            .status(500)
-            .body(json::to_vec(&err).unwrap())
-            .map_err(|e| ERRL!("{}", e));
+    let coins_ctx = try_s!(CoinsContext::from_ctx(&ctx));
+
+    // If a platform coin is to be disabled, we get all the enabled tokens for this platform coin first.
+    let tokens_to_disable = coins_ctx.get_tokens_to_disable(&ticker).await;
+    let mut tokens_and_coin_to_disable = tokens_to_disable.clone();
```

Comment:
This clone is not needed you can make `tokens_to_disable` mutable straight away and use it. Please remove `tokens_and_coin_to_disable` and use this instead.
``` rust
let mut tokens_to_disable = coins_ctx.get_tokens_to_disable(&ticker).await;
```